adapters/gumgum/gumgum.go

```diff
@@ -127,25 +119,49 @@ func (g *GumGumAdapter) MakeBids(internalRequest *openrtb.BidRequest, externalRe
 	return bidResponse, errs
 }
 
-func preprocess(imp *openrtb.Imp) (string, error) {
+func preprocess(imp *openrtb.Imp) (*openrtb_ext.ExtImpGumGum, error) {
 	var bidderExt adapters.ExtImpBidder
 	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
 		err = &errortypes.BadInput{
 			Message: err.Error(),
 		}
-		return "", err
+		return nil, err
 	}
 
 	var gumgumExt openrtb_ext.ExtImpGumGum
 	if err := json.Unmarshal(bidderExt.Bidder, &gumgumExt); err != nil {
 		err = &errortypes.BadInput{
 			Message: err.Error(),
 		}
-		return "", err
+		return nil, err
+	}
+
+	if imp.Banner != nil {
+		bannerCopy := *imp.Banner
+		if bannerCopy.W == nil && bannerCopy.H == nil && len(bannerCopy.Format) > 0 {
+			format := bannerCopy.Format[0]
+			bannerCopy.W = &(format.W)
+			bannerCopy.H = &(format.H)
+		}
+		imp.Banner = &bannerCopy
```

Comment:
Perhaps we should only copy the Banner and update the imp inside the above if statement? No need to do the work of copying the Banner if we aren't going to change anything.